server/ipqueue.go

```diff
@@ -66,10 +99,14 @@ func newIPQueue[T any](s *Server, name string, opts ...ipQueueOpt) *ipQueue[T] {
 // Add the element `e` to the queue, notifying the queue channel's `ch` if the
 // entry is the first to be added, and returns the length of the queue after
 // this element is added.
-func (q *ipQueue[T]) push(e T) int {
+func (q *ipQueue[T]) push(e T) (int, bool) {
 	var signal bool
 	q.Lock()
 	l := len(q.elts) - q.pos
+	if q.lelts > 0 && l == q.lelts {
+		q.Unlock()
+		return l, false
```

Comment:
I think just returning an error here might be better so we can distinguish between which limit we hit. WDYT?

Reply:
I did consider it but it doesn't really matter to any of the call sites, the entry doesn't get queued either way. Can add it if you prefer but may not be needed.

Reply:
We need to know which limit for the log warning IMO. We will want to rate limit them of course but we should be clear which limit we hit, bytes or number of elements.